UefiTestingPkg/AuditTests/PagingAudit/UEFI/Dxe/App/DxePagingAuditTestApp.c

```diff
@@ -29,7 +163,124 @@ DxePagingAuditTestAppEntryPoint (
   IN     EFI_SYSTEM_TABLE  *SystemTable
   )
 {
-  DumpPagingInfo (NULL, NULL);
+  EFI_STATUS                     Status;
+  UNIT_TEST_FRAMEWORK_HANDLE     Fw   = NULL;
+  UNIT_TEST_SUITE_HANDLE         Misc = NULL;
+  PAGING_AUDIT_TEST_CONTEXT      *Context;
+  IA32_CR4                       Cr4;
+  PAGING_MODE                    PagingMode;
+  IA32_MAP_ENTRY                 *Map           = NULL;
+  UINTN                          MapCount       = 0;
+  UINTN                          PagesAllocated = 0;
+  BOOLEAN                        RunTests       = TRUE;
+  EFI_SHELL_PARAMETERS_PROTOCOL  *ShellParams;
+
+  DEBUG ((DEBUG_ERROR, "%a()\n", __FUNCTION__));
+
+  DEBUG ((DEBUG_ERROR, "%a v%a\n", UNIT_TEST_APP_NAME, UNIT_TEST_APP_VERSION));
+
+  Status = gBS->HandleProtocol (
+                  gImageHandle,
+                  &gEfiShellParametersProtocolGuid,
+                  (VOID **)&ShellParams
+                  );
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "%a Could not retrieve command line args!\n", __FUNCTION__));
+    return EFI_PROTOCOL_ERROR;
+  }
+
+  if (ShellParams->Argc > 1) {
+    RunTests = FALSE;
+    if (StrnCmp (ShellParams->Argv[1], L"-r", 4) == 0) {
+      RunTests = TRUE;
+    } else if (StrnCmp (ShellParams->Argv[1], L"-h", 4) == 0) {
+      DEBUG ((DEBUG_INFO, "-h : Print available flags\n"));
+      DEBUG ((DEBUG_INFO, "-d : Dump the page table files to the EFI partition\n"));
+      DEBUG ((DEBUG_INFO, "-r : Run the application tests\n"));

```

Comment:
Should the usage mention that combining parameters is invalid? As is, it just looks at the first parameter.

Reply:
nit: We could not explicitly check for `-h` and instead dump it in the else. If the user gives any flag besides `-d` or `-r`, we just dump the help.

Reply:
Updated for both comments